From 80cd79aecb9f24a357b940802c91955f712e62e6 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Mon, 9 Nov 2020 22:29:07 +0800
Subject: [PATCH] arm64: rockchip_defconfig: Disable DRM_FBDEV_EMULATION
 temporarily

This will cause iommu fault as bellow:
Temporarily disable it before we have better solution.

[   10.015974] drm_mode_atomic_ioctl
[   10.016027] drm_atomic_set_property 0xcccccccc MODE_ID
[   10.016081] drm_atomic_set_property 0xcccccccc ACTIVE
[   10.016115] drm_atomic_set_property 0xcccccccc MODE_ID
[   10.016157] drm_atomic_set_property 0xcccccccc ACTIVE
[   10.016187] drm_atomic_set_property 0xcccccccc MODE_ID
[   10.016295] drm_atomic_set_property 0xcccccccc ACTIVE
[   10.016337] drm_atomic_set_property 0xc0c0c0c0 CRTC_ID
[   10.016390] drm_atomic_set_property 0xc0c0c0c0 CRTC_ID
[   10.016507] drm_atomic_helper_check
[   10.016716] vop2_plane_atomic_update Esmart1-win0
[   10.016781] vop2_crtc_atomic_flush iommu_enabled 1 iommu_needed :1
[   10.032142] rockchip_atomic_commit_complete
[   10.032205] drm_atomic_helper_cleanup_planes
[   10.032310] vop2_plane_cleanup_fb
[   10.032405] vop2_atomic_plane_destroy_state
[   10.032424] rockchip_gem_free_object
[   10.048738] rk_iommu fe043e00.iommu: Page fault at 0x0000000000030000
of type read
[   10.048791] rk_iommu fe043e00.iommu: iova = 0x0000000000030000:
dte_index: 0x0 pte_index: 0x30 page_offset: 0x0
[   10.048812] rk_iommu fe043e00.iommu: mmu_dte_addr: 0x000000007abd7000
dte@0x000000007abd7000: 0x7a01d001 valid: 1 pte@0x000000007
a01d0c0: 0x7a629006 valid: 0 page@0x0000000000000000 flags: 0x0
[   10.048870] [drm:rockchip_drm_fault_handler] *ERROR* iommu fault
handler flags: 0x28b
[   10.048902] Video Port0: DISABLED
[   10.048927]
[   10.048927] 0x00000000:
[   10.048931] 00008000
[   10.048956] 40158023
[   10.048981] 80000000
[   10.049008] 00000000
[   10.049034]
[   10.049034] 0x00000010:
[   10.049043] 00000ac0
[   10.049055] 00000000
[   10.049063] 00000000
[   10.049089] 00000000
[   10.049100]
[   10.049100] 0x00000020:
[   10.049122] 00000000
[   10.049147] e0000010
[   10.049172] 00010010
[   10.049196] 00000000

Change-Id: Id63a2f558f4ec096b405fad2f426dc525edc68b4
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 arch/arm64/configs/rockchip_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/configs/rockchip_defconfig b/arch/arm64/configs/rockchip_defconfig
index 32b7dcfd1a21..a9368620c038 100644
--- a/arch/arm64/configs/rockchip_defconfig
+++ b/arch/arm64/configs/rockchip_defconfig
@@ -559,6 +559,7 @@ CONFIG_VIDEO_SGM3784=y
 # CONFIG_VGA_ARB is not set
 CONFIG_DRM=y
 CONFIG_DRM_IGNORE_IOTCL_PERMIT=y
+# CONFIG_DRM_FBDEV_EMULATION is not set
 CONFIG_DRM_LOAD_EDID_FIRMWARE=y
 CONFIG_DRM_ROCKCHIP=y
 CONFIG_ROCKCHIP_ANALOGIX_DP=y
-- 
2.35.3

